src/docs/arch-cache.dox

```diff
@@ -1,13 +1,110 @@
-which contains various counters that drive statistics and information
-used for eviction.
+@section arch_cache_basics Basic operation
+
+Cached btree pages point to each other, mirroring the structure of the on-disk btree.
+When WiredTiger opens a file, it loads the root page of the btree into memory along with the first
+level of internal pages.  To lookup an entry in a btree, WiredTiger starts from the root page
+and searches the btree until finds the entry.  If WiredTiger encounters a page that is not in
+memory, it loads that page from storage and continues the search.
+
+To load a page into the cache, WiredTiger passes the page's address cookie to the
+@ref arch-block "Block Manager"
+and gets back a buffer containing the corresponding block from the underlying file.
+If necessary, WiredTiger decrypts and decompresses the block.  Then it allocates indexing
+structures to facilitate quick binary search of the keys in the page.  The first time WiredTiger
+needs to modify or insert an entry on a page, it allocates additional structures to track these
+changes.
+
+WiredTiger tracks the total amount of data in the cache.  It also tracks the space used by
+_clean_, (unmodified) pages and by _dirty_ (modified) pages.  When the cache becomes too
+full or contains too much dirty data, WiredTiger invokes @ref arch-eviction "Eviction" to
+remove data from the
+cache.  To remove a clean page from the cache, WiredTiger simply frees the page's memory.
+To remove a dirty page, WiredTiger must first _reconcile_ the page (converting it from
+in-memory format to on-disk format) and then write it to storage.
+
+@section arch_cache_structure Cache structure
+
+Internally, WiredTiger's cache state is represented by the \c WT_CACHE structure, which contains
+counters and parameter settings for tracking cache usage and controlling eviction policy.
+The \c WT_CACHE also includes state WiredTiger uses to track the progress of eviction.
+There is a single \c WT_CACHE for each connection, accessed via the \c WT_CONNECTION structure.
+
+Each page in the cache is accessed via a \c WT_REF structure.  When WiredTiger opens a btree,
+it places a \c WT_REF for the cached root page in the corresponding \c WT_BTREE structure.
+A \c WT_REF can represent either
+a page in the cache or one that has not been loaded yet.
+The page itself is represented by a \c WT_PAGE structure.  This includes a pointer to a buffer
+that contains the on-disk page image (decrypted and uncompressed).  It also holds the supplemental
+structures that WiredTiger uses to access and update the page while it is cached.
+
+When WiredTiger loads a page into the cache, it allocates an internal table with one entry
+for each entry on the page.  The type of these entries depends on the page type.  An internal
+btree page will have an array of \c WT_REF structures.  A btree leaf page will have an array
+of \c WT_ROW structures representing the KV pairs.  WiredTiger uses this table to find entries
```

Comment:
Could be WT_COL structures as well?

Reply:
Yeah.  I was simplifying a bit by skipping the column store details.  I've added a few sentences to describe the fix- and variable-length column store options.